encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt

```diff
@@ -52,18 +44,11 @@ public class AutomatticEncryptedLogging(
             encryptedLogSqlUtils,
             logEncrypter,
             preferenceUtilsWrapper,
-            dispatcher,
             EncryptedWellConfig(context)
         )
     }
 
-    @Suppress("unused")
-    @Subscribe(threadMode = ThreadMode.ASYNC)
-    internal fun onEncryptedLogUploaded(event: OnEncryptedLogUploaded) {
-        uploadState.value = event
-    }
-
-    override fun enqueueSendingEncryptedLogs(
+    override suspend fun enqueueSendingEncryptedLogs(
```

Comment:
We can do this or, to keep API contract intact, have our own Coroutine scope and allow clients to have an easier, non-coroutines API

Reply:
Yea, I was thinking about it too @wzieba, like moving this `CoroutineScope(Dispatchers.IO).launch { ... }` here (from where it was before). I was also thinking whether we want to do that for all current public `suspend` APIs, like `resetUploadStates()` and `uploadEncryptedLogs()`, wdyt? 🤔

Reply:
FYI: I just asked WCAndroid team here: p1750327131116929-slack-CGPNUU63E